importers/src/main/groovy/whelk/importer/DatasetImporter.groovy

```diff
@@ -62,24 +111,167 @@ class DatasetImporter {
         }
 
         List<String> needsRetry = []
-        long deletedCount = removeDeleted(whelk, dataset, idsInInput, needsRetry)
+        long deletedCount = removeDeleted(idsInInput, needsRetry)
 
         System.err.println("Created: " + createdCount +" new,\n" +
                 "updated: " + updatedCount + " existing and\n" +
                 "deleted: " + deletedCount + " old records (should have been: " + (deletedCount + needsRetry.size()) + "),\n" +
-                "out of the: " + idsInInput.size() + " records in dataset: \"" + dataset + "\".\n" +
+                "out of the: " + idsInInput.size() + " records in dataset: \"" + dsInfo.uri + "\".\n" +
                 "Dataset now in sync.")
     }
 
-    protected static long removeDeleted(Whelk whelk, String dataset, Set<String> idsInInput, List<String> needsRetry) {
+    void dropDataset() {
+        if (dsInfo == null) {
+            dsInfo =  new DatasetInfo([(ID): datasetUri])
+        }
+        long deletedCount = removeDeleted([] as Set, [])
+        System.err.println("Deleted dataset ${dsInfo.uri} with ${deletedCount} existing records")
+    }
+
+    protected DatasetInfo getDatasetInfo(String datasetUri, Map givenData=null) {
+        if (givenData) {
+            Map dsData = givenData
+            if (GRAPH in givenData) {
+                dsData = (Map) givenData[GRAPH].find { it[ID] == datasetUri }
+            }
+            if (dsData[ID] != datasetUri) {
+                throw new RuntimeException("Provided dataset ${givenData[ID]} does not match: ${datasetUri}")
+            }
+            return new DatasetInfo(dsData)
+        }
+        Document datasetRecord = whelk.storage.getDocumentByIri(datasetUri)
+        if (datasetRecord == null) {
+            throw new RuntimeException("Could not get dataset data for: $datasetUri")
+        }
+        Map datasetData = ((List) datasetRecord.data[GRAPH])[1]
+        assert datasetData[ID] == datasetUri
+        return new DatasetInfo(datasetData)
+    }
+
+    protected Document completeRecord(Map data, String recordType) {
+        if (GRAPH !in data) {
+            data = makeSystemRecordData(data, recordType)
+        }
+
+        if (REVERSE in data) {
+            Map revs = data.remove(REVERSE)
+        }
```

Comment:
This doesn't do anything since data is always `@graph`?

`@reverse` is always removed here anyway
https://github.com/libris/librisxl/blob/cba147b8218c7ddd8d28daf59a934339773da6a4/whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy#L2322

Reply:
Ah, right, I've cobbled up what `data` is here. Also certainly redundant; I'll remove this.